automation-tests/test/utilities.ts

```diff
@@ -0,0 +1,337 @@
+    }
+  }
+}
+
+async function notificationIsPresent(workbench: Workbench, notificationMessage: string): Promise<boolean> {
+  const notifications = await workbench.getNotifications();
+  for (const notification of notifications) {
+    const message = await notification.getMessage();
+    if (message === notificationMessage) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+async function attemptToFindNotification(workbench: Workbench, notificationMessage: string, attempts: number): Promise<boolean> {
+  while (attempts > 0) {
+    if (notificationIsPresent(workbench, notificationMessage)) {
+      return true;
+    }
+
+    pause(1);
+    attempts--;
+  }
+
+  return false;
+}
+
+async function selectChannel(outputView: OutputView, name: string): Promise<void> {
+  // Open the Output panel's dropdown menu.
+  const select = await outputView.parent.$('select.monaco-select-box');
+  await select.click();
+
+  // const channels = await outputView1.parent.$$(`${outputView1.locatorMap.BottomBarViews.outputChannels} option`);
+  const channels = await select.$$('option');
+  for (const channel of channels) {
+    const val = await channel.getValue();
+    if (val === name) {
+        await channel.click();
+        // eslint-disable-next-line wdio/no-pause
+        await browser.pause(200);
+        await browser.keys(['Escape']);
+        return;
+    }
+  }
+
+  throw new Error(`Channel ${name} not found`);
+}
+
+async function getOutputPanelText(outputChannelName: string): Promise<string> {
+  const workbench = await (await browser.getWorkbench()).wait();
+  const bottomBar = await workbench.getBottomBar(); // selector is 'div[id="workbench.parts.panel"]'
+  const outputView = await bottomBar.openOutputView(); // selector is 'div[id="workbench.panel.output"]'
+  await utilities.pause(2);
+
+  selectChannel(outputView, outputChannelName);
+  await utilities.pause(1);
+
+  // Set focus to the contents in the Output panel.
+  await (await outputView.elem).click();
+  await utilities.pause(1);
+
+  // Select all of the text within the panel.
+  await browser.keys([CMD_KEY, 'a', 'c']);
+  // Should be able to use Keys.Ctrl, but Keys is not exported from webdriverio
+  // See https://webdriver.io/docs/api/browser/keys/
+
+  const outputPanelText = await clipboard.read();
+
+  return outputPanelText;
+}
+
+// If found, this function returns the entire text that's in the Output panel
+async function attemptToFindOutputPanelText(outputChannelName: string, searchString: string, attempts: number) : Promise<string> {
+  while (attempts > 0) {
+    const outputPanelText = await getOutputPanelText(outputChannelName);
+    if (outputPanelText.includes(searchString)) {
+      return outputPanelText;
+    }
+
+    pause(1);
+    attempts--;
+  }
+
+  return undefined;
+}
+
+async function executeCommand(workbench: Workbench, command: string): Promise<TerminalView> {
```

Comment:
nit here on naming as well. I'd def assume a vscode command as the parameter here vs a system command. maybe `executeOsCommand`?